### ISSUE TITLE
Use head block instead of safe block for snap sync

### ISSUE THREAD

**Issue by jframe** [hyperledger/besu#7536]
Change snap sync to use head block instead of safe block for the snap sync.

Using the head allows for larger download windows between two pivot blocks. With larger windows, we will have fewer pivot blocks and therefore more time for the heal. This could allow us to finish faster. If we download the storage of an account and then change the pivot block, we need to request the account again to get the new storage root of the account in the block, as this can cause a lot of storage requests to fail.

Acceptance Criteria
* Verify that this has the improvement on healing and syncing that we are expecting

**Comment by kingnhcomcast** [comment]
take a look at `PivotBlockSelector`

**Comment by garyschulte** [comment]
Adding some color to the acceptance criteria, I would suggest a couple ways to quantify the benefit of head versus safe snap-sync pivot blocks

1. we can just empirically count the number of re-pivots we have when syncing a large network like mainnet, sepolia or holesky.  This misses some of the point about whether this represents a performance improvement or not, but it is a good metric to capture.  For PoS networks, the logic that controls pivot selection can be found and instrumented [here](https://github.com/hyperledger/besu/blob/main/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/DynamicPivotBlockSelector.java#L71-L151) 

2. there are two snap sync stages that should in theory benefit from fewer and longer-lasting re-pivots.  Those are the worldstate download and the worldstate heal stages.  Again as long as we are syncing a large enough network over a long enough time period, we can just observe the time it takes for each stage to complete when we are using `head` versus `safe` for the pivot block, on separate besu instances.

**Comment by kingnhcomcast** [comment]
I made changes to sync to head block and added a configuration option to revert to syncing to safe if needed: `Xsnapsync-to-head-enabled=false`

To test I ran 2 nodes at the same time one syncing to head and the other to safe.  As expected there was a drastic reduction in the number of pivots: anywhere from 11 to 50% decrease. However the actual duration to sync was minimal at just 2% decrease in time. One run was even slower but perhaps something else was going on.
